### ISSUE TITLE
more search capabilities

### ISSUE THREAD

**Issue by melyshu** [sebthedev/PrincetonCourses#80]
I'm thinking:

'1xx', '2xx', '3xx', '4xx', '5xx' to detect 100/200/300/400/500 level classes. These should OR like other fields (i.e. a search of 'cos 1xx 2xx' should return all 100 and 200 level cos courses)

'm', 'tu', 'w', 'th', 'f' or something similar to search for classes within those days. As above, these should OR like other fields so that a search of 'cos m w th' should return all cos courses whose classes are on monday, wednesday, or thursday. (probably makes sense to find classes that fit within the days rather than an exact match)

(stretch) maaaaaaybe some time restrictions: '8a', '9a', '10a', ... to represent the hours 8am-9am, 9am-10am, 10am-11am etc. These should OR like other fields, so that 't th 8a 9a 10a' should search for all classes held within 8am-11am on tuesdays and thursdays. One decision to make if implemented - do we restrict just on the lecture time or all section times (maybe put the times into course clash??)

In terms of informing the user about these search features, I'm working on the suggest pane. It will (hopefully) be intuitive how these search queries work from the way they're displayed there.

**Comment by melyshu** [comment]
more thoughts on modularizing search fields:

pdf options should all OR like other fields (i.e. 'pdfo npdf' should yield all pdf only and all npdf courses). maybe add a 'pdfx' option for those courses that don't have pdf data

audit options: include a 'naudit' for completeness' sake. OR as above. (potentially 'auditx' for those courses that don't have audit data... if there are any?)

(maybe) consider filtering by rating as well: {1.x, 2.x, 3.x, 4.x, x.x} or {1+, 2+, 3+, 4+, n/a} or similar (the current 'new' badge would fit into this category). OR as above.

**Comment by sebthedev** [comment]
* Searching by catalog number level (`1xx`, `3xx`, etc…) now implemented on the backend.
* Searching for `NAUDIT` now implemented on the backend.

**Comment by sebthedev** [comment]
Filtering based on when the course meets (either by day of the week or by time) is both complicated to implement and difficult to define what the "correct" implementation is.

If a user searches for `m w` does this mean…
* Return all courses that have have *any* class on Monday or Wednesday
* Return all courses that have have *all of their* classes on Monday or Wednesday
* Return all courses that have have *all of their* __lectures__ on Monday or Wednesday
* Or any other reasonable implementation…

Thoughts?

**Comment by melyshu** [comment]
I agree that this is not well defined... but I think the most intuitive implementation would be to return all classes that have their _main_ section(s) (lecture/seminar/whatever) _within_ the specified days. At a first glance this is what it seems like the registrar's site is doing, and it'd probably be good to implement search-by-days in a way that is consistent with their search?
